pkg/clusteragent/autoscaling/workload/controller_test.go

```diff
@@ -71,17 +71,29 @@ func newFixture(t *testing.T, testTime time.Time) *fixture {
 }
 
 func newFakePodAutoscaler(ns, name string, gen int64, creationTimestamp time.Time, spec datadoghq.DatadogPodAutoscalerSpec, status datadoghq.DatadogPodAutoscalerStatus) (obj *unstructured.Unstructured, dpa *datadoghq.DatadogPodAutoscaler) {
-	dpa = &datadoghq.DatadogPodAutoscaler{
-		TypeMeta: podAutoscalerMeta,
-		ObjectMeta: metav1.ObjectMeta{
-			Name:              name,
-			Namespace:         ns,
-			Generation:        gen,
-			UID:               uuid.NewUUID(),
-			CreationTimestamp: metav1.NewTime(creationTimestamp),
-		},
-		Spec:   spec,
-		Status: status,
+	if gen == -1 { // Create fake pod autoscaler for remote owner
```

Comment:
nit: avoid duplication and set in `dpa` if `gen > 0` (0 is not a valid value in existing object IIRC).
```
if gen > 0 {
  dpa.UID = ...
  dpa.Generation = gen
  dpa.CreationTimestamp = ...
}
```